### PR TITLE
[CALCITE-3561] Support using unnest in Interpreter

### PR DESCRIPTION
**Pull request by yanlin-Lynn**
Jira: [CALCITE-3561](https://issues.apache.org/jira/browse/CALCITE-3561)

This PR tries to add support for using unnest in `Interpreter`.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/interpreter/UncollectNode.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          Map map = (Map) value;
+          for (Object key : map.keySet()) {
+            if (rel.withOrdinality) {
+              sink.send(Row.of(key, map.get(key), i++));
+            } else {
+              sink.send(Row.of(key, map.get(key)));
+            }
+          }
+        } else {
+          throw new UnsupportedOperationException("Invalid type for unnest.");
```
Comment (asereda-gs) [inline core/src/main/java/org/apache/calcite/interpreter/UncollectNode.java]: Can you include the type of `value` in the error message so it easier to debug ?

Comment (yanlin-Lynn) [reply core/src/main/java/org/apache/calcite/interpreter/UncollectNode.java]: Thanks for review. @asereda-gs 
PR updated.

**Comment by asereda-gs** [comment]
als pls squash commits into one

File: core/src/main/java/org/apache/calcite/interpreter/UncollectNode.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+              sink.send(Row.of(key, map.get(key), i++));
+            } else {
+              sink.send(Row.of(key, map.get(key)));
+            }
+          }
+        } else {
+          throw new UnsupportedOperationException(
+              String.format(Locale.ROOT,
+                  "Invalid type: %s for unnest.",
+                  value.getClass().getCanonicalName()));
```
Comment (asereda-gs) [inline core/src/main/java/org/apache/calcite/interpreter/UncollectNode.java]: There is no risk of `value == null` ?

Comment (yanlin-Lynn) [reply core/src/main/java/org/apache/calcite/interpreter/UncollectNode.java]: Your concern is valid, it has NPE risk here.
I'll add some test to cover this.

**Comment by asereda-gs** [comment]
Thank you for PR, @yanlin-Lynn !

**Comment by yanlin-Lynn** [comment]
Thanks for review and merge, I'll delete the branch.

**Pull request merged**